l10n_ec_delivery_note/models/delivery_note.py

```diff
@@ -0,0 +1,583 @@
+from contextlib import contextmanager
+
+from odoo import api, fields, models
+from odoo.exceptions import UserError, ValidationError
+from odoo.tools.translate import _
+
+STATES = {"draft": [("readonly", False)]}
+
+
+class DeliveryNote(models.Model):
+    _inherit = ["mail.thread", "mail.activity.mixin", "portal.mixin", "sequence.mixin"]
+    _name = "l10n_ec.delivery.note"
+    _description = "Delivery Note"
+    _rec_name = "document_number"
+
+    _sequence_field = "document_number"
+    _sequence_date_field = "transfer_date"
+
+    document_number = fields.Char(
+        size=20,
+        readonly=True,
+        index=True,
+        tracking=True,
+        compute="_compute_document_number",
+        store=True,
+    )
+    journal_id = fields.Many2one(
+        comodel_name="account.journal",
+        string="Emission Point",
+        readonly=True,
+        states=STATES,
+        check_company=True,
+        domain=[("l10n_latam_use_documents", "=", True), ("type", "=", "sale")],
+    )
+    transfer_date = fields.Date(
+        required=True,
+        readonly=True,
+        states=STATES,
```

Comment:
Why do we have STATES in all this fields? I can understand probably the journal_id, but I can't understand those in transfer_date, delivery_date and so on seems to be weird to me. Could you elaborate please?